mongo/options/clientoptions.go

```diff
@@ -569,7 +569,7 @@ func (c *ClientOptions) SetMaxConnIdleTime(d time.Duration) *ClientOptions {
 
 // SetMaxPoolSize specifies that maximum number of connections allowed in the driver's connection pool to each server.
 // Requests to a server will block if this maximum is reached. This can also be set through the "maxPoolSize" URI option
-// (e.g. "maxPoolSize=100"). The default is 100. If this is 0, it will be set to math.MaxInt64.
+// (e.g. "maxPoolSize=100"). If this is 0, maximum connection pool size is not limited. The default is 100.
```

Comment:
Ah great catch. Was this previously true before your refactor?

Reply:
Yes it was true before the refactor. The previous pool implementation used a counting semaphore that either used a max of `maxPoolSize` or `math.MaxInt64`, depending on the value of `maxPoolSize`. The current pool implementation bypasses the max pool size condition if `maxPoolSize=0`. The resulting behavior is basically identical, but `math.MaxInt64` is no longer used.